libs/ml/include/ml/ops/mask_fill.hpp

```diff
@@ -40,76 +38,32 @@ class MaskFill : public Ops<T>
   using SPType        = OpMaskFillSaveableParams<TensorType>;
   using MyType        = MaskFill<TensorType>;
 
-  explicit MaskFill(DataType fill_value)
-    : fill_value_(fill_value)
-  {}
+  explicit MaskFill(DataType fill_value);
 
-  explicit MaskFill(SPType const &sp)
-    : Ops<T>(sp)
-  {
-    fill_value_ = sp.fill_value;
-  }
+  explicit MaskFill(SPType const &sp);
 
   ~MaskFill() override = default;
 
-  std::shared_ptr<OpsSaveableParams> GetOpSaveableParams() override
-  {
-    auto sp        = std::make_shared<SPType>();
-    sp->fill_value = fill_value_;
-    return sp;
-  }
+  std::shared_ptr<OpsSaveableParams> GetOpSaveableParams() override;
 
   std::shared_ptr<fetch::ml::ops::Ops<TensorType>> MakeSharedCopy(
-      std::shared_ptr<fetch::ml::ops::Ops<TensorType>> me) override
-  {
-    FETCH_UNUSED(me);
-    assert(me.get() == this);
-
-    auto copyshare = std::make_shared<MyType>(*this);  // calls default copy constructor of MyType
-
-    return copyshare;
-  }
+      std::shared_ptr<fetch::ml::ops::Ops<TensorType>> me) override;
   /**
```

Comment:
generally i preferred to move these descriptions to the cpp as well; that way the hpp is very concise and clear, and if you want more details they're all in one place (the cpp)

Reply:
Done